arangod/RocksDBEngine/CMakeLists.txt

```diff
@@ -1,107 +1,126 @@
-# -*- mode: CMAKE; -*-
+add_library(arango_rocksdb STATIC
+  RocksDBBackgroundThread.cpp
+  RocksDBBuilderIndex.cpp
+  RocksDBChecksumEnv.cpp
+  RocksDBCollection.cpp
+  RocksDBColumnFamilyManager.cpp
+  RocksDBCommon.cpp
+  RocksDBComparator.cpp
+  RocksDBCuckooIndexEstimator.cpp
+  RocksDBEdgeIndex.cpp
+  RocksDBEngine.cpp
+  RocksDBFormat.cpp
+  RocksDBFulltextIndex.cpp
+  RocksDBGeoIndex.cpp
+  RocksDBIncrementalSync.cpp
+  RocksDBIndex.cpp
+  RocksDBIndexFactory.cpp
+  RocksDBIterators.cpp
+  RocksDBKey.cpp
+  RocksDBKeyBounds.cpp
+  RocksDBLogValue.cpp
+  RocksDBMetaCollection.cpp
+  RocksDBMetadata.cpp
+  RocksDBOptimizerRules.cpp
+  RocksDBOptionFeature.cpp
+  RocksDBOptionsProvider.cpp
+  RocksDBPersistedLog.cpp
+  RocksDBPrimaryIndex.cpp
+  RocksDBRecoveryManager.cpp
+  RocksDBReplicationCommon.cpp
+  RocksDBReplicationContext.cpp
+  RocksDBReplicationIterator.cpp
+  RocksDBReplicationManager.cpp
+  RocksDBReplicationTailing.cpp
+  RocksDBRestCollectionHandler.cpp
+  RocksDBRestHandlers.cpp
+  RocksDBRestReplicationHandler.cpp
+  RocksDBRestWalHandler.cpp
+  RocksDBSavePoint.cpp
+  RocksDBSettingsManager.cpp
+  RocksDBSyncThread.cpp
+  RocksDBSortedRowsStorageContext.cpp
+  RocksDBTempStorage.cpp
+  RocksDBTransactionCollection.cpp
+  RocksDBTransactionMethods.cpp
+  RocksDBTransactionState.cpp
+  RocksDBTtlIndex.cpp
+  RocksDBTypes.cpp
+  RocksDBUpgrade.cpp
+  RocksDBV8Functions.cpp
+  RocksDBVPackIndex.cpp
+  RocksDBValue.cpp
+  RocksDBWalAccess.cpp
+  RocksDBZkdIndex.cpp
+  ReplicatedRocksDBTransactionCollection.cpp
+  ReplicatedRocksDBTransactionState.cpp
+  SimpleRocksDBTransactionState.cpp
+  SortedRowsStorageBackendRocksDB.cpp)
+
+target_link_libraries(arango_rocksdb
+  arango_cache
+  arango_common_rest_handler
+  arango_geo
+  arango_health
+  arango_indexes
+  arango_restart_action
+  arango_storage_engine_common
+  arango::validation
+  arango_zkd
+  boost_boost
+  rocksdb
+  rocksdb_interface
+  snappy)
+
+set(SNAPPY_INCLUDE_DIR "${SNAPPY_SOURCE_DIR};${SNAPPY_BUILD_DIR}" CACHE PATH "relation to snappy")
+target_include_directories(arango_rocksdb PUBLIC ${SNAPPY_INCLUDE_DIR})
+
+if (USE_ENTERPRISE)
+  # this is required for hotbackup. Views need to be flushed.
+  target_include_directories(arango_rocksdb PUBLIC
+    ${IRESEARCH_INCLUDE})
+	target_link_libraries(arango_rocksdb
+    arango_rclone)
+endif()
+
+if ((CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_CLANG) AND NOT BUILDING_FOR_ARM64)
+  target_compile_options(arango_rocksdb PUBLIC "-maes")
+endif ()
 
-##
```

Comment:
I'm not sure if these definitions are really needed: I restricted them to only the arango_rocksdb library, before they were added to all libraries in arangod.